docs/book/rest-api.md

```diff
@@ -1888,3 +1891,38 @@ OPA currently supports the following query performance metrics:
 - **timer_rego_query_parse_ns**: time taken (in nanonseconds) to parse the query.
 - **timer_rego_query_compile_ns**: time taken (in nanonseconds) to compile the query.
 - **timer_rego_query_eval_ns**: time taken (in nanonseconds) to evaluate the query.
+
+## <a name="watches"></a> Watches
+
+OPA can set watches on queries and report whenever the result of evaluating the query has changed. When a watch is set on a query, the requesting connection will be maintained as the query results are streamed back in HTTP Chunked Encoding format. A notification reflecting a certain change to the query results will be delivered _at most once_. That is, if a watch is set on `data.x`, and then multiple writes are made to `data.x`, say `1`, `2` and `3`, only the notification reflecting `data.x=3` is always seen eventually (assuming the watch is not ended, there are no connection problems, etc). The notifications reflecting `data.x=1` and `data.x=2` _might_ be seen. However, the notifications sent are guaranteed to be in order (`data.x=2` will always come after `data.x=1`, if it comes).
+
+The notification stream will not be delimited by any value; the client is expected to be able to parse JSON values from the stream one by one, recognizing where one ends and the next begins.
+
+The notification stream is a stream of JSON objects, each of which has the following structure:
+```
+{
+    "id": "<id>",
+    "value": <result>,
```

Comment:
Thinking about this a bit more...the structure should be aligned with the rest of the API.

I propose the following:

```
{
   "result": {
    "query": "<query>",
    "value": <value>,
   }
   "error": <error>
}
```

- Eventually we can support `metrics`, `explanation`, etc. at the top-level
- `error` should be an object (not a string as it's currently specified)